engine/paymail/paymail_service_client.go

```diff
@@ -0,0 +1,209 @@
+package paymail
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/mrz1836/go-cachestore"
+)
+
+const cacheKeyCapabilities = "paymail-capabilities-"
+const cacheTTLCapabilities = 60 * time.Minute
+
+// PayloadFormat is the format of the paymail payload
+type PayloadFormat uint32
+
+// Types of Paymail payload formats
+const (
+	BasicPaymailPayloadFormat PayloadFormat = iota
+	BeefPaymailPayloadFormat
+)
+
+func (format PayloadFormat) String() string {
+	switch format {
+	case BasicPaymailPayloadFormat:
+		return "BasicPaymailPayloadFormat"
+
+	case BeefPaymailPayloadFormat:
+		return "BeefPaymailPayloadFormat"
+
+	default:
+		return fmt.Sprintf("%d", uint32(format))
+	}
+}
+
+// ServiceClient is a service that aims to make easier paymail operations.
+type ServiceClient interface {
+	GetSanitizedPaymail(addr string) (*paymail.SanitisedPaymail, error)
+	GetCapabilities(ctx context.Context, domain string) (*paymail.CapabilitiesPayload, error)
+	GetP2P(ctx context.Context, domain string) (success bool, p2pDestinationURL, p2pSubmitTxURL string, format PayloadFormat)
+	StartP2PTransaction(alias, domain, p2pDestinationURL string, satoshis uint64) (*paymail.PaymentDestinationPayload, error)
+	GetPkiForPaymail(ctx context.Context, sPaymail *paymail.SanitisedPaymail) (*paymail.PKIResponse, error)
+	AddContactRequest(ctx context.Context, receiverPaymail *paymail.SanitisedPaymail, contactData *paymail.PikeContactRequestPayload) (*paymail.PikeContactRequestResponse, error)
+}
+
+type service struct {
+	cs cachestore.ClientInterface
+	pc paymail.ClientInterface
+}
+
+// NewServiceClient creates a new paymail service client
+func NewServiceClient(cs cachestore.ClientInterface, pc paymail.ClientInterface) (ServiceClient, error) {
+	if pc == nil {
+		return nil, spverrors.Newf("paymail client is required")
+	}
+	return &service{
+		cs: cs,
+		pc: pc,
+	}, nil
+}
+
+// GetSanitizedPaymail validates and returns the sanitized version of paymail address (alias@domain.tld)
+func (s *service) GetSanitizedPaymail(addr string) (*paymail.SanitisedPaymail, error) {
+	if err := paymail.ValidatePaymail(addr); err != nil {
+		return nil, err //nolint:wrapcheck // we have handler for paymail errors
+	}
+
+	sanitized := &paymail.SanitisedPaymail{}
+	sanitized.Alias, sanitized.Domain, sanitized.Address = paymail.SanitizePaymail(addr)
+
+	return sanitized, nil
+}
+
+// GetCapabilities is a utility function to retrieve capabilities for a Paymail provider
+func (s *service) GetCapabilities(ctx context.Context, domain string) (*paymail.CapabilitiesPayload, error) {
+	// Attempt to get from cachestore
+	// todo: allow user to configure the time that they want to cache the capabilities (if they want to cache or not)
+	capabilities := new(paymail.CapabilitiesPayload)
+	if s.cs != nil {
+		if err := s.cs.GetModel(
+			ctx, cacheKeyCapabilities+domain, capabilities,
```

Comment:
I see you're making the cache key twice - here and at the bottom of this function.
Consider making a local variable 
```golang
cacheKey := cacheKeyCapabilities+domain
```
and reuse it everywhere you need in the function.